src/Stratis.Bitcoin.IntegrationTests/Miners/ProofOfStakeMintCoinsSteps.cs

```diff
@@ -0,0 +1,177 @@
+        {
+            this.nodes = this.nodeGroupBuilder
+                    .CreateStratisPowMiningNode(PowMiner)
+                    .MineCoinsFast()
+                    .Start()
+                    .NotInIBD()
+                    .WithWallet(PowWallet, PowWalletPassword)
+                    .Build();
+
+            this.powSenderAddress = this.nodes[PowMiner].FullNode.WalletManager().GetUnusedAddress(new WalletAccountReference(PowWallet, PowWalletAccount));
+            var wallet = this.nodes[PowMiner].FullNode.WalletManager().GetWalletByName(PowWallet);
+            this.powSenderPrivateKey = wallet.GetExtendedPrivateKeyForAddress(PowWalletPassword, this.powSenderAddress).PrivateKey;
+        }
+
+        private void it_mines_genesis_and_premine_blocks()
+        {
+            this.sharedSteps.MinePremineBlocks(this.nodes[PowMiner], PowWallet, PowWalletAccount, PowWalletPassword);
+        }
+
+        private void mine_coins_to_maturity()
+        {
+            this.nodes[PowMiner].GenerateStratisWithMiner(100);
+            this.sharedSteps.WaitForBlockStoreToSync(this.nodes[PowMiner]);
+        }
+
+        private void a_proof_of_stake_node_with_wallet()
+        {
+            this.nodeGroupBuilder
+                    .CreateStratisPosNode(PosStaker)
+                    .MineCoinsFast()
+                    .Start()
+                    .NotInIBD()
+                    .WithWallet(PosWallet, PosWalletPassword)
+                    .Build();
+        }
+
+        private void it_syncs_with_proof_work_node()
+        {
+            this.posReceiverAddress = this.nodes[PosStaker].FullNode.WalletManager().GetUnusedAddress(new WalletAccountReference(PosWallet, PosWalletAccount));
+            var wallet = this.nodes[PosStaker].FullNode.WalletManager().GetWalletByName(PosWallet);
+            this.posReceiverPrivateKey = wallet.GetExtendedPrivateKeyForAddress(PosWalletPassword, this.posReceiverAddress).PrivateKey;
+
+            this.nodes[PosStaker].SetDummyMinerSecret(new BitcoinSecret(this.posReceiverPrivateKey, this.nodes[PosStaker].FullNode.Network));
+            this.nodeGroupBuilder.WithConnections().Connect(PowMiner, PosStaker);
+        }
+
+        private void create_tx_to_send_million_coins_from_pow_wallet_to_pos_node_wallet()
+        {
+            var context = SharedSteps.CreateTransactionBuildContext(
+                PowWallet,
+                PowWalletAccount,
+                PowWalletPassword,
+                this.posReceiverAddress.ScriptPubKey,
+                Money.COIN * 1000000,
+                FeeType.Medium,
+                101);
+
+            context.OverrideFeeRate = new FeeRate(Money.Satoshis(20000));
+
+            var sendTransaction = CreateSendTransaction();
+            this.nodes[PowMiner].FullNode.NodeService<WalletController>().SendTransaction(new SendTransactionRequest(sendTransaction.ToHex()));
+        }
+
+        private Transaction CreateSendTransaction()
```

Comment:
do you like this private method close to its usage like this, or perhaps pull to the bottom of the file?

Reply:
Actually I moved this into the step 👍